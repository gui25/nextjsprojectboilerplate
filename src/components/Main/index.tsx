import React from 'react'

const Main = () => {
  return (
    <main className="bg-gradient-to-r from-indigo-900 to-blue-900 text-white min-h-screen pb-20 md:pb-0 pt-20 md:pt-0 flex flex-col justify-center items-center">
      <div className="max-w-4xl mx-auto px-6 text-center">
        <h1 className="text-5xl md:text-7xl font-bold mb-8">
          Next Project Boilerplate
        </h1>
        <h2 className="text-xl md:text-2xl font-normal">
          The Next.js Project Boilerplate serves as a starting point for
          developing applications using Next.js. It leverages a modern tech
          stack, including TypeScript, Tailwind, Jest, React Testing Library,
          Storybook, Eslint, Prettier, and Husky.
        </h2>
      </div>
    </main>
  )
}

export default Main
