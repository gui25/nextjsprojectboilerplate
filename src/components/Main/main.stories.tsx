import React from 'react'
import { StoryFn, Meta } from '@storybook/react'
import Main from '.'

export default {
  title: 'Components/Main',
  component: Main
} as Meta

const Template: StoryFn = (args) => <Main {...args} />

export const Default: StoryFn = Template.bind({})
Default.args = {}
