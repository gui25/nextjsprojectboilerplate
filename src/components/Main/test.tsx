import React from 'react'
import { render, screen } from '@testing-library/react'
import Main from './'

test('the Main component renders correctly', () => {
  const { asFragment } = render(<Main />)

  const titleElement = screen.getByText('Next Project Boilerplate')
  expect(titleElement).toBeInTheDocument()

  const descriptionElement = screen.getByText(
    `The Next.js Project Boilerplate serves as a starting point for developing applications using Next.js. It leverages a modern tech stack, including TypeScript, Tailwind, Jest, React Testing Library, Storybook, Eslint, Prettier, and Husky.`
  )
  expect(descriptionElement).toBeInTheDocument()

  expect(asFragment()).toMatchSnapshot()
})
