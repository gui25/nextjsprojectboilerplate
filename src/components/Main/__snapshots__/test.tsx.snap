// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Main component renders correctly 1`] = `
<DocumentFragment>
  <main
    class="bg-gradient-to-r from-indigo-900 to-blue-900 text-white min-h-screen pb-20 md:pb-0 pt-20 md:pt-0 flex flex-col justify-center items-center"
  >
    <div
      class="max-w-4xl mx-auto px-6 text-center"
    >
      <h1
        class="text-5xl md:text-7xl font-bold mb-8"
      >
        Next Project Boilerplate
      </h1>
      <h2
        class="text-xl md:text-2xl font-normal"
      >
        The Next.js Project Boilerplate serves as a starting point for developing applications using Next.js. It leverages a modern tech stack, including TypeScript, Tailwind, Jest, React Testing Library, Storybook, Eslint, Prettier, and Husky.
      </h2>
    </div>
  </main>
</DocumentFragment>
`;
